starboard/xb1/shared/video_decoder_uwp.cc

```diff
@@ -52,7 +53,13 @@ bool VideoDecoderUwp::IsHardwareAv1DecoderSupported() {
   // So to avoid using av1 mft in these devices we test if
   // the current XBOX has vp9 hw decoder. If it doesn't it means
   // that there is XboxOneS/Base and we don't use mft av1 decoder.
-  return VideoDecoderUwp::IsHardwareVp9DecoderSupported() &&
+  // Additionally as experiment we exclude SeriesS & SeriesX from
+  // available hw decoders list because new sw decoder is more faster
+  // than the mft decoder.
+  const auto xb_type = starboard::shared::uwp::GetXboxType();
+  const bool exclude_hw_av1 = xb_type == starboard::shared::uwp::kXboxSeriesS ||
```

Comment:
That's a good point. We can use the reverse logic to enable hw av1 by default (use hw_av1_disabled_devices instead).

Reply:
Coming back around on this, can we take one of the following approaches:

1) Enumerate enabled devices:

```
const std::set<XboxType> hw_av1_enabled_devices = {};

...

return hw_av1_enabled_devices.contains(xb_type) && VideoDecoderUwp::IsHardwareVp9DecoderSupported() ...
```

OR

2) Enumerate disabled devices in a constant:

```
const std::set<XboxType> hw_av1_disabled_devices = { starboard::shared::uwp::kXboxSeriesS, starboard::shared::uwp::kXboxSeriesX };

...

return !hw_av1_disabled_devices.contains(xb_type) && VideoDecoderUwp::IsHardwareVp9DecoderSupported() ...
```

I personally prefer option 1 to avoid a confusing double negative (checking if something is enabled instead of NOT disabled), but my opinion on this is weak enough to accept either approach.

Reply:
Following #1 in your comment the function will look as
// static
bool VideoDecoderUwp::IsHardwareAv1DecoderSupported() {
  // XboxOneS/Base & Xbox Series S/X have av1 mft decoder, but sw av1 decoder
  // is more preferred because of performance. Xbox One X doesn't have av1
  // decoder onboard at all. So until new devices will appear the response
  // will be "false".
  return false;
}
because none of the existing devices will support hw av1 decoding.
Are you agree with such solution?

Reply:
> Following #1 in your comment the function will look as // static bool VideoDecoderUwp::IsHardwareAv1DecoderSupported() { // XboxOneS/Base & Xbox Series S/X have av1 mft decoder, but sw av1 decoder // is more preferred because of performance. Xbox One X doesn't have av1 // decoder onboard at all. So until new devices will appear the response // will be "false". return false; } because none of the existing devices will support hw av1 decoding. Are you agree with such solution?

This is true regardless of my comment and isn't what I'd like.

I want a place in our code where it is made clear exactly the set of devices we'd like to enable/disable hardware AV1 decoding. It will have the same outcome as the line this comment is on, but will be cleaner when we need to make updates in the future.

Something like

```
...

const int kXboxeOneXMaxOutputSamples = 5;

// Set of devices that shouldn't use hardware AV1 decoding.
// TODO (b/259431190): Series devices had performance issues with hardware AV1 decoders during our initial rollout.
// Disable hardware AV1 decoding in favor of the software decoder to test if performance improves.
const std::set<XboxType> hw_av1_disabled_devices = { starboard::shared::uwp::kXboxSeriesS, starboard::shared::uwp::kXboxSeriesX };

...

// static
bool VideoDecoderUwp::IsHardwareAv1DecoderSupported() {
  // XboxOneS/Base has av1 mft decoder, but sw av1 decoder
  // is more preferred for Xbox OneS/Base because of performance.
  // So to avoid using av1 mft in these devices we test if
  // the current XBOX has vp9 hw decoder. If it doesn't it means
  // that there is XboxOneS/Base and we don't use mft av1 decoder.
  return !hw_av1_disabled_devices.contains(starboard::shared::uwp::GetXboxType()) && VideoDecoderUwp::IsHardwareVp9DecoderSupported() &&
         ::starboard::shared::win32::IsHardwareAv1DecoderSupported();
}

...
```